packages/realm-server/server.ts

```diff
@@ -293,6 +293,13 @@ export class RealmServer {
       /(<meta name="@cardstack\/host\/config\/environment" content=")([^"].*)(">)/,
       (_match, g1, g2, g3) => {
         let config = JSON.parse(decodeURIComponent(g2));
+
+        if (config.defaultPublishedRealmDomain === 'localhost:4201') {
```

Comment:
Without this the Matrix test was publishing the realm to `localhost:4201`. Just overriding without checking whether it’s the default means staging and production domains will be lost.

Another approach to this would be to have this be part of realm server configuration, if that seems better, I’m happy to update to that approach!